### PR TITLE
Nav Accessibility: Expose focus public method so that caller can explicitly set focus to Nav component.

### PR DESCRIPTION
**Pull request by inateeg**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

When Nav component is rendered, there is property 'isOnTop' can by passed in when the Nav is rendered as popup.  IN SPO case, this is when in T-narrow mode, Nav is only rendered when user click on Hamburg.  Under this condition, we would like Nav get focus since it is intentionally popup opened by user action.  This PR ensure when those conditions meet, Nav focusZone get focus so user can use arrow key to navigate through nodes.

(give an overview)

#### Focus areas to test

set isOnTop to true and verify the arrow key can navigate through the nav nodes while without this property set, that is not possible.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10310)

### CODE REVIEW

**Comment by dzearing** [comment]
Also in the future please title your PRs with meaningful names, it helps identify it at a glance. :)

**Comment by inateeg** [comment]
@David Zearing<mailto:dzearing@microsoft.com> you are right the clean & right fix is for caller to explicitly calling  navComponent.focus(forceFirstElementFocus)…  I have refactor the PR so that Nav component will expose focus public api.  Please take the look.  Only thing is deadline for accessibility fix is this Friday.  This PR will need to pick up to V6 since sp-client still on V6 for a while and then it require sp-pages code change to get Nav componentRef and call focus explicitly.  Hmm even we do 60 meter dash speed, perhaps all the PR in will pass Friday deadline that is tomorrow ☹

File: packages/office-ui-fabric-react/src/components/Nav/examples/Nav.Basic.Example.tsx [hunk 1]
```diff
@@ -83,4 +84,7 @@ function _onLinkClick(ev: React.MouseEvent<HTMLElement>, item?: INavLink) {
   if (item && item.name === 'News') {
     alert('News link clicked');
   }
+  if (_nav && _nav.current) {
+    _nav.current.focus(true);
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Nav/examples/Nav.Basic.Example.tsx]: minor nit here; can you please remove this from the example? It's nice to test the functionality but makes for a weird experience for the customer when they're looking at the example. (They would click on News, and suddenly focus would move unnaturally.

Comment (inateeg) [reply packages/office-ui-fabric-react/src/components/Nav/examples/Nav.Basic.Example.tsx]: done

**Pull request merged**